clippy_lints/src/slow_vector_initialization.rs

```diff
@@ -118,6 +120,25 @@ impl<'tcx> LateLintPass<'tcx> for SlowVectorInit {
 }
 
 impl SlowVectorInit {
+    /// Given an expression, returns:
+    /// - `Some(Some(size))` if it is a function call to `Vec::with_capacity(size)`
+    /// - `Some(None)` if it is a call to `Vec::new()`
+    /// - `None` if it is neither
+    #[allow(
+        clippy::option_option,
+        reason = "outer option is immediately removed at call site and the inner option is kept around, \
+            so extracting into a dedicated enum seems unnecessarily complicated"
```

Comment:
Yeah this seems fine, it'd definitely be better if it was an optional two-variant enum though. But not too big of a deal

Reply:
it would end up as an enum that is exactly the same as the `Option<_>` enum in std, just with different variant names, but I can do that if it makes it clearer